android/src/main/java/com/reactcommunity/rndatetimepicker/RNTimePickerDialogFragment.java

```diff
@@ -51,6 +54,7 @@ static TimePickerDialog createDialog(Bundle args, Context activityContext, @Null
     boolean is24hour = DateFormat.is24HourFormat(activityContext);
 
     RNTimePickerDisplay display = RNTimePickerDisplay.DEFAULT;
+    TimePickerDialog dialog = null;
```

Comment:
instead of having this local variable that you assign later on, would it be possible to extract the logic of the `if`s and `switch` statements into a separate function that just returns the `TimePickerDialog` instance?

Reply:
Absolutely, I've just mirrored existing code as I wanted to keep consistency of approach with existing codebase. https://github.com/react-native-community/react-native-datetimepicker/blob/bdcee75420bb796a94ef010f8c751378cd8b4e05/android/src/main/java/com/reactcommunity/rndatetimepicker/RNDatePickerDialogFragment.java#L55

Reply:
I've refactored both to have same approach